### PR TITLE
Implement sendInternal function for mainchain/sidechain stores - Closes #7040

### LINKED ISSUE #7040: Implement sendInternal method in interoperability store
### Description

Create and send a cross chain message:

```
sendInternal(feeAddress, moduleID, crossChainCommandID, receivingChainID, fee, status, params):
    # Processing on the mainchain
    if ownChainAccount.ID == MAINCHAIN_ID:
        # Chain has to exist on mainchain
        if chainAccount(receivingChainID) does not exists:
            return False
        partnerChainID = receivingChainID
    # Processing on a sidechain
    else:
        # Check for direct channel
        if chainAccount(receivingChainID) does not exists:
            partnerChainID = mainchainID
        else:
            partnerChainID = receivingChainID

    # Chain must be live; This checks is always on the receivingChainID
    if not isLive(receivingChainID):
        return False
    # Chain status must be active
    if not chainAccount(partnerChainID).status == CHAIN_ACTIVE:
        return False

    # Create cross-chain message
    ccm = {
        "nonce": ownChainAccount.nonce,
        "moduleID": moduleID,
        "crossChainCommandID": crossChainCommandID,
        "sendingChainID": ownChainAccount.ID,
        "receivingChainID": receivingChainID,
        "fee": fee,
        "status": status,
        "params": params
    }

    if validateFormat(ccm) == False:
        return False

    for each module mdl for which beforeSendCCM exists:
        mdl.beforeSendCCM(feeAddress, ccm)
        if the previous call fails:
            return False

    addToOutbox(partnerChainID, ccm)
    ownChainAccount.nonce += 1
    
    return True
```

### Acceptance Criteria

- Add unit tests for sendInternal

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #7040 

### How was it solved?

[🌱 Add schema](https://github.com/LiskHQ/lisk-sdk/commit/18aca1deeb4d16aed224ca42417690191d914e06)
  
[✅ Add get/set own account in baseInteroperabilityStore](https://github.com/LiskHQ/lisk-sdk/commit/e5280315d6537aecea78b4e497fe2a3f420b164a)
  
[🌱 Add mainchain sendInternal store function](https://github.com/LiskHQ/lisk-sdk/commit/7838a9bff0a878287f427a81ceb8c1fbd833450b)
  
[🌱 Add sendInternal fucntion for sidechain store](https://github.com/LiskHQ/lisk-sdk/commit/db3a713a0cba72b1ca2be64c9f4ef0063038b8bb)
  
[🌱 Add cross chain context testing utils](https://github.com/LiskHQ/lisk-sdk/commit/ce65096fc45f336087c1f47dbda26e80cd0c00f4)
  
[✅ Add mainchain sendInternal tests](https://github.com/LiskHQ/lisk-sdk/commit/c234563eb977cac7287d0175366facc6bd0a5898)
  
[✅ Add sidechain sendInternal unit tests](https://github.com/LiskHQ/lisk-sdk/commit/e38c31cc0e2a820973c67757a4370af8ca27dca7)

### How was it tested?

`npm run test:unit interoperability` under `framework`

### CODE REVIEW

**Review by shuse2 — CHANGES_REQUESTED** [review]

File: framework/src/modules/interoperability/mainchain/store.ts [hunk 1]
```diff
@@ -37,10 +38,58 @@ export class MainchainInteroperabilityStore extends BaseInteroperabilityStore {
...
+
+		try {
+			validateFormat(ccm);
+		} catch (error) {
+			return false;
+		}
+
+		for (const mod of this._interoperableModules.values()) {
+			if (mod?.crossChainAPI?.beforeSendCCM) {
+				await mod.crossChainAPI?.beforeSendCCM(sendContext.beforeSendContext);
```
Comment (mitsuaki-u) [inline framework/src/modules/interoperability/mainchain/store.ts:85]:
```suggestion
				await mod.crossChainAPI.beforeSendCCM(sendContext.beforeSendContext);
```

File: framework/src/modules/interoperability/sidechain/store.ts [hunk 2]
```diff
@@ -27,10 +30,61 @@ export class SidechainInteroperabilityStore extends BaseInteroperabilityStore {
...
+			if (!(error instanceof NotFoundError)) {
+				throw error;
+			}
+			partnerChainID = MAINCHAIN_ID;
+		}
+		const partnerChainIDAsStoreKey = getIDAsKeyForStore(partnerChainID);
+
+		const partnerChainAccount = await this.getChainAccount(partnerChainIDAsStoreKey);
+		// Chain must be live; This checks is always on the receivingChainID
+		// Chain must be live; This check is always on the receivingChainID
```
Comment (mitsuaki-u) [inline framework/src/modules/interoperability/sidechain/store.ts:49]: Duplicate comment

Comment (mitsuaki-u) [inline framework/src/modules/interoperability/sidechain/store.ts:49]: Duplicate comment

File: framework/src/modules/interoperability/sidechain/store.ts [hunk 3]
```diff
@@ -27,10 +30,61 @@ export class SidechainInteroperabilityStore extends BaseInteroperabilityStore {
...
+
+		try {
+			validateFormat(ccm);
+		} catch (error) {
+			return false;
+		}
+
+		for (const mod of this._interoperableModules.values()) {
+			if (mod?.crossChainAPI?.beforeSendCCM) {
+				await mod.crossChainAPI?.beforeSendCCM(sendContext.beforeSendContext);
```
Comment (mitsuaki-u) [inline framework/src/modules/interoperability/sidechain/store.ts:79]:
```suggestion
				await mod.crossChainAPI.beforeSendCCM(sendContext.beforeSendContext);
```

Comment (mitsuaki-u) [inline framework/src/modules/interoperability/sidechain/store.ts:79]:
```suggestion
				await mod.crossChainAPI.beforeSendCCM(sendContext.beforeSendContext);
```

**Review by mitsuaki-u — COMMENTED** [review]
Some minor comments

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**